gns3server/handlers/api/controller/node_handler.py

```diff
@@ -59,16 +61,40 @@ class NodeHandler:
     @Route.get(
         r"/projects/{project_id}/nodes",
         parameters={
-            "project_id": "Project UUID"
+            "project_id": "Project UUID",
+            "node_type": "Node type",
+            "status": "Status",
+            "name": "Node name",
+            "locked": "Locked"
         },
         status_codes={
             200: "List of nodes returned",
         },
         description="List nodes of a project")
     async def list_nodes(request, response):
+        project_id = request.match_info["project_id"]
+        node_type = request.query.get("node_type", None)
+        status = request.query.get("status", None)
+        name = request.query.get("name", None)
+        locked = request.query.get("locked", None)
 
-        project = await Controller.instance().get_loaded_project(request.match_info["project_id"])
-        response.json([v for v in project.nodes.values()])
+        project = await Controller.instance().get_loaded_project(project_id)
+        nodes = [v for v in project.nodes.values()]
+
+        if node_type is not None:
+            nodes = [node for node in nodes if node["node_type"] == node_type]
+
+        if status is not None:
+            nodes = [node for node in nodes if "status" in node and node["status"] == "started"]
```

Comment:
After choosing sort by status I get the following error:
Uncaught exception detected: <class 'TypeError'>
Traceback (most recent call last):
    await func (request, response) node_handler.py", line 88, in <listcomp>
    nodes = [node for node in nodes if "status" in node and node ["status"] == "started"]
TypeError: argument of type 'Node' is not iterable

Reply:
This is error I get on server side:

```
Traceback (most recent call last):
  File "/home/grossmj/PycharmProjects/gns3-server/gns3server/web/route.py", line 198, in control_schema
    await func(request, response)
  File "/home/grossmj/PycharmProjects/gns3-server/gns3server/handlers/api/controller/node_handler.py", line 88, in list_nodes
    nodes = [node for node in nodes if "status" in node and node.status == status]
  File "/home/grossmj/PycharmProjects/gns3-server/gns3server/handlers/api/controller/node_handler.py", line 88, in <listcomp>
    nodes = [node for node in nodes if "status" in node and node.status == status]
TypeError: argument of type 'Node' is not iterable
```

Notice the `<listcomp>`, I think this is the problem. I don't really know why it happens though, I rarely use list comprehensions in Python.

Also, I think you should use node.status property instead of node ["status"].